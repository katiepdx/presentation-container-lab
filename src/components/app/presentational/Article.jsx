// Display an article's title, author, and description
// Needs a snapshot test

import React from 'react';
import propTypes from 'prop-types';

// deconstruct props
const Article = ({ title, author, description }) => (
  <div>
    <h2>{title}</h2>
    <h6>{author}</h6>
    <p>{description}</p>
  </div>
)

// prop types 
Article.propTypes = {
  title: propTypes.string.isRequired,
  author: propTypes.string,
  description: propTypes.string
};

// export function component 
export default Article;
