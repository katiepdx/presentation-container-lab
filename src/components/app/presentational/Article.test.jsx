// Snapshot the article component using test info 

import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Article from './Article'

describe('Article component', () => {
  afterAll(() => cleanup());

  it('renders Article', () => {
    // snapshot fragment and render Article (pass test props)
    const { asFragment } = render(<Article
      title="Cool Article"
      author="Cool Author"
      description="Cool article description goes here"
      />);
      // expect that the snapshot matches the asFragment
      expect(asFragment()).toMatchSnapshot();
  })
});
