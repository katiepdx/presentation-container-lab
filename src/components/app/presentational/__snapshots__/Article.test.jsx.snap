// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component renders Article 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Cool Article
    </h2>
    <h6>
      Cool Author
    </h6>
    <p>
      Cool article description goes here
    </p>
  </div>
</DocumentFragment>
`;
